thinc/layers/residual.py

```diff
@@ -1,17 +1,39 @@
-from typing import Tuple, Callable, Optional, List, TypeVar
+from typing import Tuple, Callable, Optional, List, TypeVar, cast, Union
```

Comment:
Here the old version actually looks more like what we want than the new one, but with the latest mypy and pydantic versions it yields a slew of errors.